src/order.rs

```diff
@@ -0,0 +1,247 @@
+//! The maker creates an order that defines how much he wants to buy for the amount he is selling.
+//! order's buy amount = what the maker wants from a taker
+//! order's sell amount = what the maker is offering to a taker
+//!
+//! mid_market_rate is set as 1 sell => x buy, where x is the mid_market_rate
+//!
+//! BTC-DAI: When selling 1 BTC we should buy 9000 DAI, mid_market_rate is 1:9000
+//! Given BTC:DAI and the rate of 1:9000
+//!     selling 1.0 BTC with spread_pc of 3% => buy 9270 DAI
+//!     selling 0.5 BTC with spread_pc of 3% => buy 4635 DAI
+//! Given DAI:BTC and a rate of 1:0.0001
+//!     selling 10000 DAI with spread_pc of 3% => buy 1.03 BTC
+//!     selling 1000 DAI with spread_pc of 3% => buy 0.103 DAI
+//!
+
+use crate::bitcoin;
+use crate::dai;
+use crate::{Rate, Spread};
+use std::cmp::min;
+
+#[derive(Debug, Clone)]
+pub struct BtcDaiOrder<P> {
+    pub position: P,
+    pub base: bitcoin::Amount,
+    pub quote: dai::Amount,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct Buy;
+
+#[derive(Debug, Clone, Copy)]
+pub struct Sell;
+
+impl BtcDaiOrder<Buy> {
+    pub fn new<W, B>(
+        wallet: W,
```

Comment:
It might actually be better to just pass the amounts here (instead of the full wallet).
The way main event look is currently designed the wallet might not be passed around.

The wallet is a specific ethereum wallet at this point. I think we can remove this abstraction for now and make it simpler by just passing in the Amounts directly (same for the book).